src/main/java/com/ericsson/ei/cache/SubscriptionCacheHandler.java

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright 2021 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.ei.cache;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections4.map.HashedMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.handlers.DateUtils;
+
+/**
+ * This class is used for caching the subscriptions and checking the aggregated
+ * objects from the collection.
+ *
+ */
+@Component
+public class SubscriptionCacheHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionCacheHandler.class);
+    public static Map<String, List<String>> subscriptionsCache = new HashedMap<>();
+
+    // ttl value to cleanup the subscription cache. default to 1hour.
+    private final static String delayString = "3600000";
```

Comment:
How often is this subscription cache updated with new subcsriptions and subscription changes from MongoDb?
Is that happen sametime cache is cleaned?

I see this only one customer specific request, it need to be more general user implement for all kind of user.
Some users/customer maybe don't have 100000+ events per couple hour and don't have the load problem and want to have direct reactive system, when subscriptions is updated/changed, notfication should be made directly based on these changes, e.g. for changing and triggering pipelines.

I guess you can this implementation for maintenance branch and 2.x release. In Master branch this cache should be configurable and it need to be documented how this cache works and  what it solves and how to configure it.

No values should be hard coded in Master branch, it has been a lot of discussions and issues/problems with hardcoded values in other components that its not suits all use cases and all users/customer -> need to be configurable.
But this can be handled when implemented in Master branch.

Reply:
This caused me to think also about some other tools that uses EI and in System test Subscriptions is updated/changed and if the change is not used directly in the cache,, then these tests will fail.
But I have understood exactly how and when this cache re-synched with MongoDb.
How often is the subscription cache synched?

Reply:
Right, we will check on when we merge to master branch for these changes.